app/controllers/concerns/body_class_concern.rb

```diff
@@ -0,0 +1,32 @@
+module BodyClassConcern
+  extend ActiveSupport::Concern
+  extend Mandate::Memoize
+
+  memoize
+  def body_class
+    classes = []
+    classes << "devise" if devise_controller?
+    classes << "namespace-#{namespace_name}"
+    classes << "controller-#{controller_name || 'none'}"
+    classes << "action-#{action_name}"
```

Comment:
No, it can't. I have no idea under what scenario `controller_name` can be empty either, but this bit of code has followed me around for a decade so I'll trust some past version of myself on it!